### ISSUE TITLE
npm audit

### ISSUE THREAD

**Issue by Pprime1** [Pprime1/csasm#34]
ran `npm audit` got the following report. Heroku is claiming a simple command will fix these two issues ... but is it that simple for me? At what impact to my program? And do I apply this to both csasm as well as the production code ASMHUB?

_More Research to understand_
```
 === npm audit security report ===                                             
# Run  npm install socket.io@4.4.1  to resolve 1 vulnerability
  High            Uncaught Exception in engine.io                               
  Package         engine.io                                                     
  Dependency of   socket.io                                                     
  Path            socket.io > engine.io                                         
  More info       https://github.com/advisories/GHSA-273r-mgr4-v34f             

# Run  npm update ws --depth 3  to resolve 1 vulnerability
  Moderate        ReDoS in Sec-Websocket-Protocol header                        
  Package         ws                                                            
  Dependency of   socket.io                                                     
  Path            socket.io > engine.io > ws                                    
  More info       https://github.com/advisories/GHSA-6fc8-4gx4-v693             
                                                                                
found 2 vulnerabilities (1 moderate, 1 high) in 116 scanned packages
  run `npm audit fix` to fix 2 of them.
```

**Comment by Pprime1** [comment]
```
+ socket.io@4.4.1
added 1 package from 1 contributor, removed 1 package and updated 12 packages in 2.761s

fixed 2 of 2 vulnerabilities in 116 scanned packages
```

in csasm console

**Comment by Pprime1** [comment]
same in asmhub console

nothing seems broken so far

**Comment by Pprime1** [comment]
the vulnerability alerts are still there in fresh code builds?

**Comment by judsona** [comment]
These changes must be applied to the codebase. The command it recommends is meant to be run on your own pc… it looks to have updated the packages file. 

Your best bet would be to clone the repository locally, run the command (requires other tools to be installed) commit the changes locally and push to GitHub. 

Or identify which packages it updated and apply them manually

**Comment by Pprime1** [comment]
could get most done on the github website - but not the primary issue. 
```socket.io@4.0.1 requires engine.io@~5.0.0```
It would have updated engine.io to v5.2.1, but it doesn't seem to be able to install socket.io v4.4.1 first. Having never successfully "cloned the repository locally" I will need to look at editing the package-lock.json file manually. This section would be where to start
```
    "socket.io": {
      "version": "4.0.1",
      "resolved": "https://registry.npmjs.org/socket.io/-/socket.io-4.0.1.tgz",
      "integrity": "sha512-g8eZB9lV0f4X4gndG0k7YZAywOg1VxYgCUspS4V+sDqsgI/duqd0AW84pKkbGj/wQwxrqrEq+VZrspRfTbHTAQ==",
      "requires": {
        "@types/cookie": "^0.4.0",
        "@types/cors": "^2.8.8",
        "@types/node": ">=10.0.0",
        "accepts": "~1.3.4",
        "base64id": "~2.0.0",
        "debug": "~4.3.1",
        "engine.io": "~5.0.0",
        "socket.io-adapter": "~2.2.0",
        "socket.io-parser": "~4.0.3"
      },
      "dependencies": {
        "debug": {
          "version": "4.3.1",
          "resolved": "https://registry.npmjs.org/debug/-/debug-4.3.1.tgz",
          "integrity": "sha512-doEwdvm4PCeK4K3RQN2ZC2BYUBaxwLARCqZmMjtF8a51J2Rb0xpVloFRnCODwqjpwnAoao4pelN8l3RJdv3gRQ==",
          "requires": {
            "ms": "2.1.2"
          }
        },
        "ms": {
          "version": "2.1.2",
          "resolved": "https://registry.npmjs.org/ms/-/ms-2.1.2.tgz",
          "integrity": "sha512-sGkPx+VjMtmA6MX27oA4FBFELFCZZ4S4XqeGOXCv68tT+jb3vk/RyaKWP0PTKyWtmLSM0b+adUTEvbs1PEaH2w=="
        }
      }
    },
```   
... that looks a little scary to manually edit.

**Comment by judsona** [comment]
You need to look at the package.json and the package lock. 

Package lock is to ensure your production versions are consistent. Package is the original configuration file. Package lock can be regenerated

**Comment by Pprime1** [comment]
OK. Bit the bullet and installed 'github desktop', cloned a repository and a branch (yeah that's a thing). and then dumped out to a command prompt. because I did previously install some stuff last year - git cmd and heroku (I think that was it) I assumed console commands would work still.
Ran `npm audit` and got the same message as I was getting in Heroku console (plus a message that npm itself is an old version - problem for another day). So yes, that seemed to be all in place still.
So ran `npm audit fix` and it seems to have done the update of engine.io, and consequently socket.io ... says 185 packages added and 186 up to date. package-lock.json appears to be double the length. I can see that the version numbers of engine.io and socket.io are updated.
Didn't have to update package.json since I had previous manually updated socket.io in there ... so probably that's ok now.
Committed all this back to main and then committed it all back to 'origin' (ie: online github). At which point it updated the website and kicked off a build.
Which worked. Build v1058
csasm app seems to be working still. Will run my tests later.

comments above are for future reference for when I might need to do this all again.
